src/lib/structures/fpa_rounding.ml

```diff
@@ -87,8 +88,15 @@ let hstring_ae_reprs =
 
 (* The rounding mode is the enum with the SMT values.
    The Alt-Ergo values are injected in this type. *)
-let fpa_rounding_mode =
-  Ty.Tsum (Hs.make "RoundingMode", hstring_smt_reprs)
+let fpa_rounding_mode_dty, d_cstrs, fpa_rounding_mode =
+  let module DStd = Dolmen.Std in
+  let ty_cst = DE.Ty.Const.mk (DStd.Path.global "RoundingMode") 0 in
+  let cstrs =
+    List.map (fun c -> DStd.Path.global @@ to_smt_string c, []) cstrs
+  in
+  let _, cstrs = DE.Term.define_adt ty_cst [] cstrs in
+  DE.Ty.apply ty_cst [], cstrs,
+  Ty.Tsum (Uid.of_dolmen ty_cst, List.map (fun (c, _) -> Uid.of_dolmen c) cstrs)
```

Comment:
Note: I think this could be replaced with `DStd.Expr.Ty.roundingMode`. I don't think it is entirely trivial to do so, but might be worth a note in a comment.

Reply:
I didn't the enum type is shipped with `Dolmen`. We can replace it easily. It's exactly the same code.

Reply:
Uhm actually `roundingMode` is the SMT-LIB roundingMode, so it's so obvious. I add a note.